### PR TITLE
(prometheus) instant query support

### LINKED ISSUE #5765: [Feature Request] capability to choose between Prometheus "query" and "query_range" API for table
Grafana: 3.1.1 (docker image)
Prometheus: 1.0.1 (docker image)

I have the following Prometheus query:

topk(5, sum(rate(jaxrs_client_proxy_latency_seconds_sum[1h])) by (proxy, method))

When I run this in the Prometheus console I get a nice and simple "top 5"-style table because it is using the "/api/v1/query" endpoint on Prometheus. Now when I go to Grafana and add a "Table" and put the same query in I end up with a time series that I don't want/need because it is using the "/api/v1/query_range".

It would be great if - at least for the table element - it would be possible to choose what endpoint to use on Prometheus so that I can decide if I want to deal with a time series result or a simple table. Not only would it display the data that I am looking for but the "/api/v1/query" endpoint is responding about 10 times faster to the query than the "/api/v1/query_range" endpoint.

### PR DESCRIPTION
**Pull request by mtanda**
Fix https://github.com/grafana/grafana/issues/5765

### CODE REVIEW

**Comment by bergquist** [comment]
Would it make sense to have this option as a checkbox instead? Instant queries could be used for both metrics and table results I guess? 

ping @SuperQ

**Comment by SuperQ** [comment]
Yes, a checkbox totally makes sense.

File: public/app/plugins/datasource/prometheus/datasource.ts
Comment (juliusv) [inline public/app/plugins/datasource/prometheus/datasource.ts]: The Prometheus terms are "range query" vs. "instant query" - it might make sense to follow that terminology here for consistency. Both return data for time series, but the instant query just returns a single point for each "series" :)

Comment (mtanda) [reply public/app/plugins/datasource/prometheus/datasource.ts]: Oops, sorry for inconsistence.
Finally, I revert this line, and change to use checkbox, property name is `instant`.

**Comment by mtanda** [comment]
Now, support table format, and use checkbox to switch the query type.

File: public/app/plugins/datasource/prometheus/datasource.ts
Comment (bergquist) [reply public/app/plugins/datasource/prometheus/datasource.ts]: @juliusv I agree. It makes sense to use the same vocabulary as much as possible.

@mtanda great! please use the gf-form-switch component for checkboxes. Ex https://github.com/grafana/grafana/blob/master/public/app/plugins/panel/graph/tab_legend.html#L26

File: public/app/plugins/datasource/prometheus/partials/query.editor.html
Comment (bergquist) [inline public/app/plugins/datasource/prometheus/partials/query.editor.html]: @mtanda great! please use the gf-form-switch component for checkboxes. Ex https://github.com/grafana/grafana/blob/master/public/app/plugins/panel/graph/tab_legend.html#L26

changing the value should also trigger at new request.

**Review by bergquist — CHANGES_REQUESTED** [review]

File: public/app/plugins/datasource/prometheus/partials/query.editor.html
Comment (mtanda) [reply public/app/plugins/datasource/prometheus/partials/query.editor.html]: Thanks for the check! I fixed it.

**Review by bergquist — APPROVED** [review]

File: public/app/plugins/datasource/prometheus/partials/query.editor.html
Comment (bergquist) [inline public/app/plugins/datasource/prometheus/partials/query.editor.html]: One last remark :) I would have fixed it myself but we currently don't merge to master. 

Please make `label-class="width-4"` `label-class="width-5"`

After this the PR is ready for merge.

Comment (mtanda) [reply public/app/plugins/datasource/prometheus/partials/query.editor.html]: changed :)

**Comment by torkelo** [comment]
This is just an alternative approach, not sure if its better than checkbox. 

image

the question is if we need Instant as Time Series. The singlestat panel supports table format now.

**Comment by mtanda** [comment]
The instant time series query could be useful for pie chart/gauge panel.

**Comment by bergquist** [comment]
@torkelo I'm merging this. 

I see format option as a data transform option once the data get back from Prometheus. Instant queries is an option when sending queries. I think its preferable to keep them separated. I'm not that worried that we bloat the UI since metric controllers are collapsible in the metric tab. 

Also.. I need this for improving the bundled Prometheus dashboard for internal grafana metrics :)

**Comment by torkelo** [comment]
sound good

**Pull request merged**
